teloxide-macros/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+extern crate proc_macro;
+extern crate syn;
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{DeriveInput, parse_macro_input};
+
+#[proc_macro_derive(TelegramCommandEnum)]
+pub fn derive_telegram_command_enum(tokens: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(tokens as DeriveInput);
+
+    let (variant, variant_str) = match &input.data {
+        syn::Data::Enum(data) => {
+            (data.variants.iter(),
+             data.variants.iter().map(|variant| {
+                variant.ident.to_string().to_lowercase()
+            }))
+        }
+        _ => panic!("TelegramCommandEnum allowed only for enums")
```

Comment:
Macros should return something like `quote! { compile_error!("omg") }` on error so compiler  can show line where error occured

Reply:
fixed